### ISSUE TITLE
@stylexjs/eslint-plugin: Support all CSS color formats

### ISSUE THREAD

**Issue by robinlahtinen** [facebook/stylex#420]
### Describe the issue

Stylex ESLint disallows `oklch(0 0 0)` or following color formats.

### Expected behavior

Stylex ESLint allows all color formats supported by CSS Color Module Level 4 or higher.

### Steps to reproduce

1. Create a style with color property
2. Use value outside the type, for example `oklch(0 0 0)`.
3. Try linter

### Test case

```tsx
const styles = stylex.create({
    example: {
        color: "oklch(0 0 0)",
    },
});
```

**Comment by yousoumar** [comment]
The ESLint rule seems to already support any string as we have: `const color = makeUnionRule(isString, isNamedColor, isHexColor)`. Plus, I'm not getting any error with the given example with the latest version of the plugin, which is `0.5.1`.

**Comment by nedjulius** [comment]
> The ESLint rule seems to already support any string as we have: [`const color = makeUnionRule(isString, isNamedColor, isHexColor)`](https://github.com/facebook/stylex/blob/main/packages/eslint-plugin/src/stylex-valid-styles.js#L151). Plus, I'm not getting any error with the given example with the latest version of the plugin, which is `0.5.1`.

I think the OP wants to have linting support for color formats such as `oklch`, `rgba` etc. based on the current implementation, eslint just treats these formats as strings and does not report any problem with invalid values.

**Comment by yousoumar** [comment]
Sorry, in a hurry I misunderstood I guess. Thanks @nedjulius !  I want to work on this. Feel free to assign it to me @nmn.

**Comment by nmn** [comment]
A bigger improvement to the ESLint rule is planned. The plan is to use a CSS value parser to validate all CSS properties according to the specification and remove the need to accept arbitrary strings when specific patterns are expected.

I have worked on a custom CSS parser for this use-case since `postcss-value-parser` didn't do what we needed. The parser can be found [here](https://github.com/nmn/css-value-parser).

We will get back to working on this once the most urgent issues have been resolved.

**Comment by yousoumar** [comment]
So should we wait for the parser before working on this issue @nmn? As a matter of fact, I started looking at it, implemented already a validation for `rgb`, `rgba`,  `hsl` and `hsla`. They are working fine with normal strings, but not with [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals). Firstly I thought we should only have normal strings as the [doc says](https://stylexjs.com/docs/learn/styling-ui/defining-styles/#constraints), but I found one template literal in the [Next.js example app](https://github.com/facebook/stylex/blob/main/apps/nextjs-example/components/Card.tsx#L55).

**Comment by nmn** [comment]
@yousoumar Please wait for the parser to start working on this. There is a bigger refactor needed for the ESLint plugin to consistent parse complex values that are currently parsed in the Babel plugin.

Once this refactor is done, the validation logic will be implemented in vanilla JS with no AST dependency. This is similar to how the `shared` package contains all the core logic for the StyleX Babel plugin and operates on vanilla strings, objects and arrays.

**Comment by hyperknot** [comment]
Just wanted to bump this, the eslint plugin still offers to split:

```
    borderColor: 'oklch(0.675 0.050 271.237)',
```

into:

```
    borderTopColor: 'oklch(0.675',
    borderRightColor: '0.050',
    borderBottomColor: '271.237)',
    borderLeftColor: '0.050',
```
